edx_event_bus_kafka/internal/producer.py

```diff
@@ -191,14 +191,14 @@ def send(
               string naming the dictionary keys to descend)
             event_data: The event data (kwargs) sent to the signal
         """
+        full_topic = get_full_topic(topic)
+
         event_key = extract_event_key(event_data, event_key_field)
         headers = {EVENT_TYPE_HEADER_KEY: signal.event_type}
 
         key_serializer, value_serializer = get_serializers(signal, event_key_field)
-        key_bytes = key_serializer(event_key, SerializationContext(topic, MessageField.KEY, headers))
-        value_bytes = value_serializer(event_data, SerializationContext(topic, MessageField.VALUE, headers))
-
-        full_topic = get_full_topic(topic)
+        key_bytes = key_serializer(event_key, SerializationContext(full_topic, MessageField.KEY, headers))
```

Comment:
Hah, whoops! I even remember thinking "man, I don't like having `topic` mean two different things in our code" so I'm surprised I didn't switch these over too.

Reply:
Speaking of which, I wonder if we should clarify the docstring to note that the topic parameter is the "base topic", without an environment prefix.

Reply:
that makes sense, will update